src/compiler/transformers/module/module.ts

```diff
@@ -1790,7 +1790,7 @@ namespace ts {
                         const name = importDeclaration.propertyName || importDeclaration.name;
                         return setTextRange(
                             factory.createPropertyAccessExpression(
-                                factory.getGeneratedNameForNode(importDeclaration.parent.parent.parent),
+                                factory.getGeneratedNameForNode(importDeclaration.parent?.parent?.parent || importDeclaration),

```

Comment:
This change (and the similar ones in `system`) are what fix the crash mentioned in the OP. The non-module file has a reference to a synthetic import that, since it's never _actually_ inserted into the file (since it's not a module), is never parented, and thus has no name. We emit a (admittedly type system, rather than grammar/program) error in these cases usually (`module "react/jsx-runtime" not found`), so the emitted JS being nonfunctional is _probably_ fine, but this case might need a more descriptive error added. The emit simply uses an arbitrary name now (`_a`), rather than crashing.

Since this part fixes a crash, I'm wondering if I should (partially?) port this to the 4.1 branch or no? @DanielRosenwasser ?

Reply:
[This comment](https://github.com/microsoft/TypeScript/issues/41616#issuecomment-735690427) may actually be referring to the crash fixed here (assuming the input file is exactly what the user in that comment says, odd as a top-level `return` is), though the stack is very different than the on the OP of that thread refers to (and is definitely a separate issue).